File: leetcode/199.BinaryTreeRightSideView/02.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    
    vector<int> rightSideView(TreeNode* root) {
        vector<int> ans;
        if (root == nullptr) return ans;
        int maxLevel = 0;
        this->treeLevelOrder(root, 1, maxLevel, ans);
        return ans;
    }
    
    void treeLevelOrder(TreeNode* root, int level, int &maxLevel, vector<int> &ans) {
        if (level > maxLevel) {
            maxLevel++;
            ans.push_back(root->val);
        }
        if (root->right != nullptr) this->treeLevelOrder(root->right, level+1, maxLevel, ans);
        if (root->left != nullptr) this->treeLevelOrder(root->left, level+1, maxLevel, ans);
        return;
    }
};

File: leetcode/62.UniquePaths/02.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [1] * n
        ans = 0
        for i in range(m):
            for j in range(n):
                if i == 0 or j == 0:
                    ans = 1
                else:
                    ans = ans + dp[j]
                    dp[j] = ans
        return ans
